scenarios/kubernetes_e2e.py

```diff
@@ -476,6 +481,25 @@ def main(args):
         runner_args.append(
             '--gcp-service-account=%s' % mode.add_service_account(args.service_account))
 
+    if args.use_shared_build is not None:
+        # find shared build location from GCS
+        gcs_path = args.gcs_shared
+        gcs_path += os.getenv('PULL_REFS', '') + '/'
+        if args.use_shared_build:
+            gcs_path += args.use_shared_build + '-'
+        gcs_path += 'build-location.txt'
+        try:
+            build_loc = read_gcs_path(gcs_path)
+            # tell kubetest to extract from this location
+            args.kubetest_args.append('--extract=' + build_loc)
+            args.build = None
+        except urllib2.URLError as err:
+            print >>sys.stderr, 'Failed to get shared build location: %s'%err.reason
+            print >>sys.stderr, 'Falling back to local build...'
```

Comment:
why fallback? this makes things complicated and may result in the job timing out instead.